container-core/src/main/java/com/yahoo/container/handler/threadpool/ContainerThreadPool.java

```diff
@@ -83,7 +83,9 @@ private static int computeMaximumThreadPoolSize(int maxNumThreads) {
     }
 
     private static int computeCoreThreadPoolSize(int corePoolSize, int maxNumThreads) {
-        return Math.min(corePoolSize, maxNumThreads);
+        return corePoolSize <= 0
+            ? Runtime.getRuntime().availableProcessors() * 2
+            : Math.min(corePoolSize, maxNumThreads);
```

Comment:
I think that you need to do Math.min after you computed corePoolSize again.

Like
{code}
if (corePoolSize <= 0) corePoolSize = Runtime.getRuntime().availableProcessors() * 2 : corePoolSize;
Math.min(corePoolSize, maxNumThreads);
{code}